src/isolate_setup.cpp

```diff
@@ -243,3 +257,16 @@ Dart_Isolate CreateIsolate(bool is_main_isolate,
 
   return isolate;
 }
+
+void* GetCallbackData(void* isolate_data) {
+  if (isolate_data == nullptr) {
+    return nullptr;
+  }
+  IsolateGroupData* isolate_group_data = static_cast<IsolateData*>(isolate_data)->isolate_group_data();
+  
+  if (isolate_group_data == nullptr) {
+    return nullptr;
+  }
+  void* callback_data = static_cast<DllIsolateGroupData*>(isolate_group_data)->callback_data;
```

Comment:
I don't think this double cast is necessary. You should be able to cast directly to `DllIsolateGroupData` since this isn't a dynamic cast.

Reply:
`isolate_group_data()` returns `IsolateGroupData*` so we need to cast back to the parent type to access `callback_data` unless there is another way im missing here.

Reply:
No you're right. Sorry, when I initially read this I though you cast `isolate_data` directly to the group data.